### DISCUSSION TITLE
Mafiles for ASF

### DISCUSSION THREAD

**Discussion by DeverBa** [dyc3/steamguard-cli#394] [Q&A] [answered]
Is it possible to use the mafiles created by SteamGuard CLI to import them into ASF (Archi Steam Farm)? I decided to ask this because ASF itself is a very handy tool, and I would like to have this option.

**Comment by dyc3** [comment]
It's probably technically possible, as both tools should use the same APIs for logging in. But as of right now, there's no functionality in steamguard-cli to export to ASF's format.

I'd be open to PRs for the functionality.

**Comment by JustArchi** [comment, marked as answer]
According to @dyc3 **[here](https://github.com/dyc3/steamguard-cli/discussions/388#discussioncomment-9852271)**, the program's format of the maFile is:

```
{
  "account_name": "example",
  "steam_id": 1234,
  "serial_number": "kljasfhds",
  "revocation_code": "R12345",
  "shared_secret": "zvIayp3JPvtvX/QGHqsqKBk/44s=",
  "token_gid": "jkkjlhkhjgf",
  "identity_secret": "kjsdlwowiqe=",
  "uri": "otpauth://totp/Steam:example?secret=ASDF&issuer=Steam",
  "device_id": "android:99d2ad0e-4bad-4247-b111-26393aae0be3",
  "secret_1": "sklduhfgsdlkjhf=",
  "tokens": null,
}
```

ASF requires a valid json file with only `identity_secret` and `shared_secret` fields. If your maFile has those two, i.e. looks similar to above, then it can be imported to ASF right away with no required edits or additional details.

Ref: https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Two-factor-authentication#advanced
